### ISSUE TITLE
Popups and ComponentListeners

### ISSUE THREAD

**Issue by TheBiblMan** [mgarin/weblaf#277]
ComponentListener.componentMoved only seems to fire when the popup is first created.
https://github.com/TheBiblMan/Java-Bytecode-Wrecker/blob/development/src/main/java/org/topdank/jbcwr/ui/features/search/SearchModel.java#L220
https://github.com/TheBiblMan/Java-Bytecode-Wrecker/blob/development/src/main/java/org/topdank/jbcwr/ui/features/search/SearchPopup.java#L24
(Taken partially from the 1.28 demo)

Edit: Instead of displaying the popup by passing a WebFrame instance, I passed a WebPanel instance and suddenly there are more movement events however they are not being called frequently enough. (Moved as in when the containing WebFrame moves and so then does the WebPanel).

**Comment by mgarin** [comment]
I am not yet sure what you are trying to do with that listener by looking at the code...

---

Though, let me give a bit of information on `WebPopup` and `ComponentListener`.

`WebPopup` is basically a simple panel that is made "popup" by `PopupManager` which places it at absolute coordinates relative to window root pane in a window's layered pane. Since it is placed inside the layered pane it's display area is also strictly limited by root pane bounds.

`ComponentListener` is one of base Swing listeners which informs about component location and visibility changes. Note that locations here means location inside parent component. Also note that visibility changes here means only "visible" component property change, not the actual component visibility on screen - basically it is called when you call `component.setVisible ( true / false )`.

So, combining this information - if you add `ComponentListener` to `WebPopup` - `componentMoved` will fire only once when component is shown because it is placed inside the layered pane under new coordinates which is basically a "move" event. It might also fire again if component you have shown this popup for is moved inside of your application (popup will react to that and move accordingly) or if you manually change popup location/bounds.

So basically, it doesn't seem to be an issue. Though again, as I said in the beginning, - I'm not sure what you are trying to do there so I can't give any good advice how it can be done.

**Comment by mgarin** [comment]
Also, probably you might want to try out `WebPopOver` instead of `WebPopup`. It is a completely separate dialog-based custom-styled window that displays your content. It can also be re-styled to meet your requirements.

**Comment by TheBiblMan** [comment]
Thanks, I'll try it out.

**Comment by mgarin** [comment]
Feel free to reopen this issue if you still cannot achieve your goal.
I'll try to help out, though I will need a bit more information on the case.
